PRASCore.jl/src/Systems/assets.jl

```diff
@@ -1,6 +1,38 @@
 abstract type AbstractAssets{N,L,T<:Period,P<:PowerUnit} end
 Base.length(a::AbstractAssets) = length(a.names)
 
+function Base.show(io::IO, asset_collection::T) where {T<:AbstractAssets}
+
+    # Count occurrences of each category
+    category_counts = Dict{String, Int}()
+    for category in asset_collection.categories
```

Comment:
This assumes people aren't inputting garbage values in the category which I guess is a good assumption.